src/sentry/eventstream/kafka.py

```diff
@@ -18,30 +19,45 @@
 
 # Beware! Changing this, or the message format/fields themselves requires
 # consideration of all downstream consumers.
+EVENT_PROTOCOL_VERSION = 2
+
 # Version 1 format: (1, TYPE, [...REST...])
 #   Insert: (1, 'insert', {
 #       ...event json...
 #   }, {
 #       ...state for post-processing...
 #   })
-#   Delete Groups: (1, 'delete_groups', {
+#
+#   Mutations that *should be ignored*: (1, ('delete_groups'|'unmerge'|'merge'), {...})
+#
+#   In short, for protocol version 1 only messages starting with (1, 'insert', ...)
+#   should be processed.
+
+# Version 2 format: (2, TYPE, [...REST...])
+#   Insert: (2, 'insert', {
+#       ...event json...
+#   }, {
+#       ...state for post-processing...
+#   })
+#   Delete Groups: (2, '(start_delete_groups|end_delete_groups)', {
 #       'project_id': id,
-#       'group_ids': [id1, id2, id3],
+#       'group_ids': [id2, id2, id3],
 #       'datetime': timestamp,
```

Comment:
Thanks, I think it probably makes sense to avoid using the message timestamp in that case and send it as part of the payload that we explicitly control.